### ISSUE TITLE
Reverting to older version of a package in Pubspec doesn't actually cause any changes.

### ISSUE THREAD

**Issue by ThinkDigitalRepair** [flutter/flutter#22097]
## Steps to Reproduce

1. Enter a pub dependency like 
2. `simple_permissions: ^0.1.6`
3. upgrade to `simple_permissions: ^0.1.7` by editing the pubspec and running packages get
4. edit line back to `simple_permissions: ^0.1.6` and run flutter packages get
5. Notice that file changes are not reflected. 
6. Suffer.

## Logs
NA

```
```

```
```

```
flutter doctor -v
[✓] Flutter (Channel beta, v0.8.2, on Mac OS X 10.14 18A389, locale en-US)
    • Flutter version 0.8.2 at /Users/thinkdigital/development/flutter
    • Framework revision 5ab9e70727 (13 days ago), 2018-09-07 12:33:05 -0700
    • Engine revision 58a1894a1c
    • Dart version 2.1.0-dev.3.1.flutter-760a9690c2

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.1)
    • Android SDK at /Users/thinkdigital/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.1
    • ANDROID_HOME = /Users/thinkdigital/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Ultimate Edition (version 2018.2.3)
    • IntelliJ at /Users/thinkdigital/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 26.0.3
    • Dart plugin version 182.3569.4

[✓] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.18.0

[✓] Connected devices (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android P (API 27) (emulator)

• No issues found!
```

**Comment by ThinkDigitalRepair** [comment]
as I'm reading, I think this is how pub is supposed to work. How do I tie it to one specific version? I have a package I think broke my application and I'm trying to downgrade it

**Comment by ThinkDigitalRepair** [comment]
Found it. By restraining the dependency like this, it works `package: '<=/0.1.6'`

**Comment by zoechi** [comment]
`^1.2.3` allows all versions `1.x.x`. 

Below `1.0.0` it is a bit different.
`^0.1.2` allows all version `0.1.x`.

This matches the semantic versioning rules Dart is using where above 1.0.0 the major version should be incremented for breaking changes and below 1.0.0 the minor version number should be incremented for breaking changes.

You can use fixed versions like `1.2.3` or custom ranges like `">= 1.2.3 < 1.3.0"`

**Comment by ThanksILoveYouWithAllTheRespect** [comment]
this works for me '<=1.4.0'
